e2e/test/scenarios/dashboard/reproductions/12926-editing-dashboards-doesnt-cancel-running-queries.cy.spec.js

```diff
@@ -0,0 +1,80 @@
+import {
+  editDashboard,
+  restore,
+  showDashboardCardActions,
+  undo,
+} from "e2e/support/helpers";
+
+const questionDetails = {
+  name: "Q1",
+  native: { query: "SELECT 1" },
+};
+
+describe("issue 12926", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  describe("card removal while query is in progress", () => {
+    it("should stop the ongoing query when removing a card from a dashboard", () => {
+      slowDownQuery();
+
+      cy.createNativeQuestionAndDashboard({
+        questionDetails,
+      }).then(({ body: { dashboard_id } }) => {
+        cy.visit(`/dashboard/${dashboard_id}`);
+      });
+
+      cy.window().then(win => {
+        cy.stub(win.XMLHttpRequest.prototype, "abort").as("xhrAbort");
+      });
+
+      removeCard();
+
+      cy.get("@xhrAbort").should("have.been.calledOnce");
+    });
+
+    it("should re-fetch the query when doing undo on the removal", () => {
+      slowDownQuery();
+
+      cy.createNativeQuestionAndDashboard({
+        questionDetails,
+      }).then(({ body: { dashboard_id } }) => {
+        cy.visit(`/dashboard/${dashboard_id}`);
+      });
+
+      removeCard();
+
+      restoreQuery();
+
+      undo();
+
+      cy.wait("@cardQueryRestored");
+    });
+  });
+});
+
+function slowDownQuery() {
+  cy.intercept("POST", "api/dashboard/*/dashcard/*/card/*/query", req => {
+    req.on("response", res => {
+      res.setDelay(10000);
+    });
+  }).as("cardQuerySlowed");
+}
+
+function restoreQuery() {
+  cy.intercept("POST", "api/dashboard/*/dashcard/*/card/*/query", req => {
+    req.continue();
+  }).as("cardQueryRestored");
```

Comment:
I'm not sure what `req.continue()` does in this case. Would leaving this option out entirely still work? e.g.
```suggestion
  cy.intercept("POST", "api/dashboard/*/dashcard/*/card/*/query").as("cardQueryRestored");
```

Reply:
You're right that it's not immediately clear what it does, passing a new "handler" it's actually needed to override all previous intercepts: https://github.com/cypress-io/cypress/discussions/17051#discussioncomment-1042965

Do you think I should add a comment or is the function small enough that it's self explanatory?

Reply:
Thanks for the explanation. This might not be obvious at first, but a small comment should help make things clearer.